edx-solutions/edx-platform
# [MCKIN-8184] Create an API to list course enrollments by course or a list of users

Creates an enrollment API endpoint to list course enrollments by course or a list of users. 

**Testing instructions**
* Create some test courses and users.
* Enroll some users at random into the created courses.
* Create a staff user (`is_staff` set to `True`).
* Login as the staff user.
* The endpoint is `/api/enrollment/v1/enrollments` with at least of one `course_id`, `username` query string parameters required. The `course_id` field accepts a single URL-encoded course ID and the `username` field accepts a string containing comma-separated usernames, URL encoded if necessary. Refer to the [view docstring](https://github.com/open-craft/edx-platform/blob/MCKIN-8184-api-to-return-list-of-course-enrollments-by-course-or-users/common/djangoapps/enrollment/views.py#L713) for more details.
* Test and verify the endpoint using the following example requests
```
GET /api/enrollment/v1/enrollments?course_id={course_id}

GET /api/enrollment/v1/enrollments?username={username},{username},{username}

GET /api/enrollment/v1/enrollments?course_id={course_id}&username={username}
```

**Reviewers**
